### PR TITLE
Per level naming

### LINKED ISSUE #25: Ability to use bem-config per level

### PR DESCRIPTION
**Pull request by awinogradov**
Close #25

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -52,7 +52,7 @@ On file system:
 
 ```
 blocks/Attach/
-├── Attach.react.js
```
Comment (Yeti-or) [inline README.md]: this was added to show how techMap works:
check it earlier in readMe
```
     techMap: {
         js : ['react.js']
     }
```

File: index.js [hunk 2]
```diff
@@ -4,8 +4,9 @@ const path = require('path'),
     bn = require('@bem/naming'),
     BemCell = require('@bem/cell'),
     BemEntityName = require('@bem/entity-name'),
-    bemFs = require('@bem/fs-scheme')(),
+    bemFs = require('@bem/fs-scheme')(), // TODO: plain
```
Comment (Yeti-or) [inline index.js]: Maybe it's better to add issue instead of TODO ?

File: index.js [hunk 3]
```diff
@@ -17,12 +18,9 @@ module.exports = function(source) {
...
 
     const callback = this.async(),
-        options = Object.assign(
-            {},
-            this.options.bemLoader,
-            loaderUtils.getOptions(this)
-        ),
-        levels = options.levels,
+        options = Object.assign({}, this.options.bemLoader, loaderUtils.getOptions(this)),
+        levelsMap = bemConfig.levelMapSync() || opts.levels,
```
Comment (Yeti-or) [inline index.js]: So then change readme to show that levels could be hash

File: index.js [hunk 4]
```diff
@@ -68,7 +67,8 @@ module.exports = function(source) {
         }, [])
         // find path for every entity and check it existance
         .map(bemCell => {
-            const entityPath = path.resolve(bemFs.path(bemCell, namingOptions));
+            const localNamingOpts = levelsMap[bemCell.layer].naming || namingOptions;
```
Comment (Yeti-or) [inline index.js]: I don't sure but I think that bemCell.layer is absolute path. did you check it?

File: package.json [hunk 5]
```diff
@@ -15,18 +15,19 @@
...
+    "@bem/entity-name": "^1.5.0",
     "@bem/fs-scheme": "^2.1.0",
-    "@bem/import-notation": "^1.0.0",
-    "@bem/naming": "^2.0.0-3",
-    "falafel": "^1.2.0",
-    "loader-utils": "1.0.2",
+    "@bem/import-notation": "^1.1.2",
+    "@bem/naming": "^2.0.0-6",
+    "bem-config": "^3.2.3",
+    "falafel": "^2.1.0",
```
Comment (Yeti-or) [inline package.json]: it's major, nothing changed in API ?

File: index.js (hunk 4 above)
Comment (awinogradov) [inline index.js]: Yes, it's ok

File: package.json (hunk 5 above)
Comment (awinogradov) [inline package.json]: It works)

File: index.js [hunk 6]
```diff
@@ -53,6 +51,7 @@ module.exports = function(source) {
 
         const existingEntitiesPromises = bemImport.parse(
             node.arguments[0].value,
+            // FIXME: we really need this context for parsing import?
```
Comment (awinogradov) [inline index.js]: It's my comment) I ask about next line)

File: README.md (hunk 1 above)
Comment (awinogradov) [inline README.md]: Ok, I'll revert this changes

File: index.js (hunk 6 above)
Comment (Yeti-or) [inline index.js]: It's could be a problem if we use bemNaming here
I suppose we need to use bem-fs-scheme here because we are parsing name of file

Comment (Yeti-or) [inline index.js]: Depends on https://github.com/bem-sdk/bem-fs-scheme/issues/18

**Pull request merged**